TombEngine/Game/control/trigger.cpp

```diff
@@ -152,6 +151,10 @@ int SwitchTrigger(short itemNumber, short timer)
 	}
 	else if (item->Status)
 	{
+		if (item->ObjectNumber == ID_LOCK_SWITCH &&
```

Comment:
I don't understand the purpose of this code. Can you provide an example how it is used (video + TE setup screenshot + description)?

Reply:
Maybe look more fussy explained than done.

The object can be taken from Tomb Raider 5, Joby 3 (submarine level), there are already with name AIRLOCK_SWITCH copied to TEN is also copied in slot 820 also named AIRLOCK_SWITCH.
![image text](https://i.imgur.com/JvsWT1O.jpg)

This is the image of the setup in editor:
![image text](https://i.imgur.com/R2stw37.jpg)

You put the lock and the door in the same sector, and put triggers for both in the same sector too. (For the lock, the trigger must be SWITCH type). No OCB are required.

This is how it works in game:
[![Lock Switch Text](https://img.youtube.com/vi/0i7i8ehkIlg/0.jpg)](https://www.youtube.com/watch?v=0i7i8ehkIlg)


The segment of code you marked is to activate the trigger. The normal switches when activate, they do their animation and then get disabled. When disconnected, is when they activate their trigger. (That's why anicommand "Kill Entity" is required in the transition animations in normal switches).

However, the valve on the door has two animations. (The Lara interaction and the switch opening with the door). Originally the asset hasn't got any "Kill Entity" so it doesn't trigger anything with switches code.

- If put "Kill Entity" in first animation, it does trigger after the Lara's interaction animation, but in TEN, that "disable" the object, so although it triggers the door, it doesn´t perform the next switch animation.
- If put the "Kill Entity" in the second animation, (door opening animation), the switch does the two animations, but only do the trigger at the end of two animations, what is too late.

That code is overriding that step, doing the check to activate trigger while are in STATE 1. (first frame of Animation 2).

Adittionally I added its own control and collision function, although are very similar to generic switches, I removed many lines that were not used in this escenerio, besides, generic switches IDs are OFF 1, ON 0, but for this door lock are OFF 0, ON 1 So instead to make another exception like the Jump Switch, I went with its own code what makes it more autonomous.